dockstore-integration-testing/src/test/java/io/dockstore/common/CommonTestUtilities.java

```diff
@@ -45,34 +45,99 @@ private CommonTestUtilities() {
 
     }
 
-    public static void dropAndRecreate(DropwizardTestSupport<DockstoreWebserviceConfiguration> support) throws Exception {
+    /**
+     * Drops the database and recreates from migrations, not including any test data, using new application
+     * @param support
+     * @throws Exception
+     */
+    public static void dropAndRecreateNoTestData(DropwizardTestSupport<DockstoreWebserviceConfiguration> support) throws Exception {
         Application<DockstoreWebserviceConfiguration> application = support.newApplication();
         application.run("db", "drop-all", "--confirm-delete-everything", CONFIG_PATH);
         application.run("db", "migrate", CONFIG_PATH, "--include", "1.3.0.generated,1.4.0");
     }
 
-    public static void cleanState(DropwizardTestSupport<DockstoreWebserviceConfiguration> support) throws Exception {
-        getTestingPostgres().clearDatabase();
+    /**
+     * Drops the database and recreates from migrations for non-confidential tests, using existing application
+     * @param support
+     * @throws Exception
+     */
+    public static void cleanStateWithTestData(DropwizardTestSupport<DockstoreWebserviceConfiguration> support) throws Exception {
+        support.getApplication().run("db", "drop-all", "--confirm-delete-everything", CONFIG_PATH);
+        support.getApplication().run("db", "migrate", CONFIG_PATH, "--include", "1.3.0.generated");
         support.getApplication().run("db", "migrate", CONFIG_PATH, "--include", "test");
+        support.getApplication().run("db", "migrate", CONFIG_PATH, "--include", "1.4.0");
+    }
+
+    /**
+     * Drops the database and recreates from migrations for non-confidential tests, using new application
+     * @param support
+     * @throws Exception
+     */
+    public static void dropAndCreateWithTestData(DropwizardTestSupport<DockstoreWebserviceConfiguration> support) throws Exception {
+        Application<DockstoreWebserviceConfiguration> application = support.newApplication();
```

Comment:
is this identical to cleanStateWithTestData?

Reply:
No, the cleanStateWithTestData does not create a new application for support, instead it does support.getApplication()

Reply:
Ah, ok